go-apps/meep-tc-engine/ip-mgr.go

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2019  InterDigital Communications, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	log "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-logger"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const IP_ADDR_NONE = "n/a"
+const DEFAULT_TICKER_INTERVAL_MS = 1000
+
+type IpAddrUpdateCb func()
+
+// IpManager - IP address manager
+type IpManager struct {
+	name        string
+	sandboxName string
+	podIpMap    map[string]string
+	svcIpMap    map[string]string
+	ticker      *time.Ticker
+	clientset   *kubernetes.Clientset
+	updateCb    IpAddrUpdateCb
+	isConnected bool
+	mutex       sync.Mutex
+}
+
+// NewIpManager - Creates and initialize an IP Manager instance
+func NewIpManager(name string, sandboxName string, updateCb IpAddrUpdateCb) (im *IpManager, err error) {
+
+	// Create new IP Manager instance
+	im = new(IpManager)
+	im.name = name
+	im.sandboxName = sandboxName
+	im.updateCb = updateCb
+	im.podIpMap = make(map[string]string)
+	im.svcIpMap = make(map[string]string)
+
+	log.Info("Successfully create IP Manager")
+	return im, nil
+}
+
+// SetPodList - Set list of pods to monitor IP addresses for
+func (im *IpManager) SetPodList(podList map[string]bool) {
+	im.mutex.Lock()
+	defer im.mutex.Unlock()
+
+	// Remove stale entries
+	for podName := range im.podIpMap {
+		if _, found := podList[podName]; !found {
+			delete(im.podIpMap, podName)
+		}
+	}
+
+	// Add missing entries
+	for podName := range podList {
+		if _, found := im.podIpMap[podName]; !found {
+			im.podIpMap[podName] = IP_ADDR_NONE
+		}
+	}
+}
+
+// SetSvcList - Set list of services to monitor IP addresses for
```

Comment:
This function performs an "Update" / "Refresh" operation rather than "Set" & should be named accordingly

Reply:
Modified as suggested